### PR TITLE
Migrate api-keys view to dashboard

### PR DESCRIPTION
**Pull request by rubenmoya**
### Description
This PR  migrates the API keys view (`/your_apps`) in order to decouple the dashboard from editor. Taking advantage of this change we also implemented the [new API Keys UI](https://projects.invisionapp.com/share/P2DZ1FWUE#/screens/258570869).

### Files migrated
You can check the list of migrated files [here](https://docs.google.com/spreadsheets/d/1ylS7hXamWP8dQ9ZgOAIVEvfNb2XAFY9Y2tErhDKtLUg).

### New files
- [x] `dashboard/components/form-components/editors/multi-checkbox/multi-checkbox.js`
- [x] `dashboard/components/table-grants/table-grants.js`
- [x] `dashboard/data/api-key-model.js`
- [x] `dashboard/data/api-keys-collection.js`
- [x] `dashboard/data/user-tables-model.js`
- [x] `dashboard/views/api-keys/alert-dialog-view.js`
- [x] `dashboard/views/api-keys/api-keys-form-view.js`
- [x] `dashboard/views/api-keys/api-keys-list-item-view.js`
- [x] `dashboard/views/api-keys/api-keys-list-view.js`

### Acceptance steps
- [ ] New view is shown only if the user has `builder` active and `new_dashboard` feature flag.
- [ ] When visiting `/your_apps` check you have `master` and `public` keys by default.
- [ ] Regenerate key in `master` works
- [ ] You can create new API keys
- [ ] You can regenerate regular API keys
- [ ] You can delete regular API keys
- [ ] You can search datasets in the new API key form
- [ ] When you have a Google API key it's shown after your API keys

### Screenshots
screen shot 2018-02-19 at 10 50 17
screen shot 2018-02-19 at 10 50 25
screen shot 2018-02-19 at 10 50 33
screen shot 2018-02-19 at 10 50 39
screen shot 2018-02-19 at 10 50 53

### CODE REVIEW

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -140,7 +140,7 @@ def stylesheet_link_tag *sources
   def raise_on_asset_absence *sources
     sources.flatten.each do |source|
       next if source == {:media => "all"}
-      raise "Hey, #{source} is not in the precompile list. This will fall apart in production." unless Rails.application.config.assets.precompile.any? do |matcher|
+      raise "Hey, #{source} is not in the precompile list (check application.rb). This will fall apart in production." unless Rails.application.config.assets.precompile.any? do |matcher|
```
[the remark this answers is not in the document]
Comment (rubenmoya) [reply app/helpers/application_helper.rb]: I haven't added 66 new characters cmon!

**Comment by jesusbotella** [comment]
We need to take into account that this PR branch has another generated file named `user_feed_new_vendor.js` that is included in `public_user_feed` template. Do not know if it will fail when merging.

**Comment by jesusbotella** [comment]
⚠️ 🚨 Ask @javitonino or @amartinj before merging this branch🚨⚠️ 

This branch has a backend fix to allow us to test in staging that needs to be sorted out

**Comment by amartinj** [comment]
Hey! Don't merge this PR. We need to merge [this one](https://github.com/CartoDB/cartodb/pull/13597) first. It properly fixes the "hack" @javitonino added to allow you do the acceptance. You can finish the acceptance using the branch as it is. Once you're done and think it should be merged, let us know, so we can revert his fix and merge it

**Comment by rubenmoya** [comment]
https://github.com/CartoDB/cartodb/pull/13597 has been merged so we're good to go!

**Comment by jesusbotella** [comment]
🎉

**Pull request merged**